### PR TITLE
Remove parts of the API that is no longer publicly documented 

### PR DESCRIPTION
**Pull request by ramya-stripe**
This PR contains the changes generated after removing the shared overrides from our code generator that were added for backcompat

## Changelog

* Remove the support for the below deprecated values in `BalanceTransaction.Type`
    * `obligation_inbound`
    * `obligation_payout`
    * `obligation_payout_failure`
    * `obligation_reversal_outbound`
* Remove the below deprecated events from `Event.Type`, `WebhookEndpointCreateOptions.EnabledEvent`, `WebhookEndpointUpdateOptions.EnabledEvent`
   * `invoiceitem.updated`
   * `order.created`
   * `recipient.created`
   * `recipient.deleted`
   * `recipient.updated`
   * `sku.created`
   * `sku.deleted`
   * `sku.updated`
* Remove support for `id_bank_transfer`, `multibanco`, `netbanking`, `pay_by_bank`, and `upi` on `PaymentMethodConfiguration` by removing the below classes
   * `PaymentMethodConfiguration.IdBankTransfer`
   * `PaymentMethodConfiguration.Multibanco`
   * `PaymentMethodConfiguration.Netbanking`
   * `PaymentMethodConfiguration.PayByBank`
   * `PaymentMethodConfiguration.Upi`
* Remove the support for `challenge_only` in `SetupIntent.PaymentMethodOptions.Card.RequestThreeDSecure`
* Remove the support for deprecated value `service_tax` in `TaxRate.TaxType`, `InvoiceLinetItemUpdateParams.TaxAmount.TaxRateData.TaxType`,pweb`TaxRateCreateParams.TaxType`, `TaxRateUpdateParams.TaxType`
* Remove the support for `various` in `Climate.Supplier.removalPathway` 
* Remove the deprecated value `INCLUDE_AND_REQUIRE` on the enum `InvoiceCreateParams.PendingInvoiceItemsBehavior`
* Remove the property `RequestIncrementalAuthorization` on `PaymentIntentConfirmParams.PaymentMethodOptions.CardPresent`, `PaymentIntentCreateParams.PaymentMethodOptions.CardPresent` and `PaymentIntentUpdateParams.PaymentMethodOptions.CardPresent`. This was shipped by mistake.
* Remove the support for deprecated value `obligation` on `ReportRunCreateParams.ReportingCategory`

### CODE REVIEW

File: src/main/java/com/stripe/model/PaymentSource.java [hunk 1]
```diff
@@ -1,12 +1,4 @@
...
-import com.stripe.exception.StripeException;
-import com.stripe.net.RequestOptions;
-import java.util.Map;
-
-public interface PaymentSource extends StripeObjectInterface, HasId {
-  PaymentSource update(Map<String, Object> params, RequestOptions options) throws StripeException;
-
-  PaymentSource update(Map<String, Object> params) throws StripeException;
-}
+public interface PaymentSource extends StripeObjectInterface, HasId {}
```
Comment (ramya-stripe) [inline src/main/java/com/stripe/model/PaymentSource.java:4]: Not sure why this change was generated

**Review by richardm-stripe — APPROVED** [review]

File: src/main/java/com/stripe/model/PaymentSource.java (hunk 1 above)
Comment (ramya-stripe) [reply src/main/java/com/stripe/model/PaymentSource.java:4]: Ah this was part of https://github.com/stripe/sdk-codegen/pull/1083

**Comment by ramya-stripe** [comment]
Compat check failures are expected as we are making breaking changes
